### PR TITLE
Added the Shortest Path Finder gssoc'21

### LINKED ISSUE #496: Shortest Path Finder.
# Aim

- A shortest pathfinder between two points. 

# Details

- The user can add a path manually.
- The user can add the start point and endpoint manually.
- The shortest path is displayed in the GUI with a color.
- The area tracked can also be seen in the GUI.

### PR DESCRIPTION
**Pull request by SpecTEviL**
# Description

Path-Finder is a Shortest Path Finding GUI application made with Pygame. It finds the shortest path between two points having some obstacles and thus avoiding them. It also shows the area of the board tracked to find the shortest path.

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by kaustubhgupta** [comment]
Please review @santushtisharma10 @tamanna-verma @antrikshmisri as I don't have much experience in this algo. PIng me when it's ready to be merged, will assign the tags

**Review by antrikshmisri — CHANGES_REQUESTED** [review]
@SpecTEviL Looks good , just handle this error which occurs at boundaries
Screenshot (162)

**Comment by SpecTEviL** [comment]
> @SpecTEviL Looks good , just add handle this error which occurs at boundaries
> Screenshot (162)

Its not occurring in my local machine. Can you retry that? Or else elaborate on the error? @antrikshmisri

**Comment by SpecTEviL** [comment]
> > @SpecTEviL Looks good , just add handle this error which occurs at boundaries
> > Screenshot (162)
> 
> Its not occurring in my local machine. Can you retry that? Or else elaborate on the error? @antrikshmisri



> @SpecTEviL Looks good , just add handle this error which occurs at boundaries
> Screenshot (162)

Also, can you recheck if it's my project only? Because the file you're running has a different name than mine. So just check that once and if the error occurs again do let me know. I'll debug it. @antrikshmisri 
Thanks!

**Comment by antrikshmisri** [comment]
@SpecTEviL Yes , it is your script just under different name. I think the problem is in your grid placement.
<br/>
Screenshot (163)
As you can see near the black block there is still some space to place a block and as soon as I place a block there an `IndexError` is thrown

**Comment by santushtisharma10** [comment]
@SpecTEviL I agree with @antrikshmisri .The problem is in your grid placement.

File: Shortest-Path-Finder (GUI)/Shortest Path Finder.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
+# Click on boxes to mark start and end point. Then make some obstacles and once done
+# press SPACE to find the shortest path between the start and end point
+
+import pygame as pg
+import math
+from queue import PriorityQueue
+
+
+WIDTH = 600
+WIN = pg.display.set_mode((WIDTH, WIDTH))
```
Comment (santushtisharma10) [inline Shortest-Path-Finder (GUI)/Shortest Path Finder.py:10]: Change the width to the multiple of your rows(e.g. change the width to 595)

**Comment by SpecTEviL** [comment]
> @SpecTEviL I agree with @antrikshmisri .The problem is in your grid placement.



> @SpecTEviL I agree with @antrikshmisri .The problem is in your grid placement.

Ok, I'll debug it and get back to it.
Thank you @antrikshmisri and @santushtisharma10 for letting me know about this error. 
I'll debug it ASAP!

File: Shortest-Path-Finder (GUI)/Shortest Path Finder.py (hunk 1 above)
Comment (SpecTEviL) [reply Shortest-Path-Finder (GUI)/Shortest Path Finder.py:10]: > Change the width to the multiple of your rows(e.g. change the width to 595)

Sure I'll do it ASAP. Thanks for the suggestion.

**Comment by SpecTEviL** [comment]
> > @SpecTEviL I agree with @antrikshmisri .The problem is in your grid placement.
> 
> > @SpecTEviL I agree with @antrikshmisri .The problem is in your grid placement.
> 
> Ok, I'll debug it and get back to it.
> Thank you @antrikshmisri and @santushtisharma10 for letting me know about this error.
> I'll debug it ASAP!

Hey, @santushtisharma10 and @antrikshmisri I've made the required changes. You can review it now. If the error still persists do let me know. 
Thanks!

**Review by santushtisharma10 — APPROVED** [review]

**Comment by santushtisharma10** [comment]
@antrikshmisri @tamanna-verma

**Review by antrikshmisri — APPROVED** [review]
@SpecTEviL LGTM 🚀

**Comment by avinashkranjan** [comment]
@all-contributors please add @SpecTEviL for code and documentation

**Pull request merged**